Data Structures/Heaps/heap.cpp

```diff
@@ -0,0 +1,116 @@
+#include<iostream>
+#include<vector>
+using namespace std;
+
+class MinHeap{
+    void heapify(int index){
+        //finding the index of left and right child respectively for comparison
+        int left_child = 2 * index;
+        int right_child = left_child +1;
+        //Setting min_index to first index
+        int min_index = index;
+        //Checking whether left or right child has a value which is less than value at current min_index
+        if(left_child < v.size() && v[left_child] < v[min_index] ){
+            min_index = left_child;
+        }
+        if(right_child < v.size() && v[right_child] < v[min_index]){
+            min_index = right_child;
+        }
+        //Checking whether element at index index <=={(note: no typo} contains the minimum element or not
+        if(min_index != index){
+            swap(v[min_index],v[index]);
+            heapify(min_index);
+        }
+    }
+        public:
+            vector<int> v;
+            //Constructor
+            MinHeap(){
+                //Setting v[0] to any garbage considering only positive elements inside heap
+                v.push_back(-1);
+            }
+
+            //building the heap
+            void buildHeap(int data){
```

Comment:
This function should be called as **insertToHeap** since you are inserting one element into the heap. So, if you insert N elements into heap, you need to call it `N` times and thus the complexity would be `O(NlogN)`. 
Try to write another function **buildHeap** that takes an entire integer array of `N` elements and builds heap for all the elements in one go in complexity `O(N)`.